src/tiled/wangdock.cpp

```diff
@@ -333,13 +431,31 @@ void WangDock::setCurrentWangSet(WangSet *wangSet)
     emit currentWangSetChanged(mCurrentWangSet);
 }
 
+void WangDock::activateErase()
+{
+    mEraseWangIdsButton->setChecked(true);
+
+    mCurrentWangId = 0;
+
+    mWangTemplateView->selectionModel()->clearCurrentIndex();
+    mWangTemplateView->selectionModel()->clearSelection();
+    mWangColorView->selectionModel()->clearCurrentIndex();
+    mWangColorView->selectionModel()->clearSelection();
+    emit currentWangIdChanged(mCurrentWangId);
+}
+
 void WangDock::retranslateUi()
 {
     setWindowTitle(tr("Wang Sets"));
 
     mEraseWangIdsButton->setText(tr("Erase WangIds"));
     mAddWangSet->setText(tr("Add Wang Set"));
     mRemoveWangSet->setText(tr("Remove Wang Set"));
+
+    if (mWangColorView->isVisible())
+        mSwitchTemplateViewButton->setText(tr("Switch to Template View"));
```

Comment:
Maybe you could try to use a `QTabWidget` and just have "Templates" and "Colors" tabs? It just feel like this would be a more natural way than having this toggle button. Though if you want to speed up the merging of this patch, feel free to just change the minimum based on my feedback for now and leave this for later.

Also, it occurred to me that maybe having two things called "Templates" would be confusing (Mohamed is also working on adding templates, after all). So maybe we can find a different name for this view...